.github/workflows/ci_tests.yml

```diff
@@ -46,59 +49,92 @@ jobs:
           - description: "ASan"
             args: "-DCMAKE_CXX_FLAGS='-fsanitize=address -fsanitize=undefined'"
         include:
-          - platform: ubuntu-latest
-            compiler:
+          - platform:
+              description: "ubuntu gcc"
               cpp: g++
               c: gcc
+              os: ubuntu-latest
             cpp_version: 17
             cmake_args:
               description: "Werror"
               args: "-DCMAKE_CXX_FLAGS='-Werror=all -Werror=extra'"
-          - platform: ubuntu-latest
-            compiler:
+          - platform:
+              description: "ubuntu gcc"
               cpp: g++
               c: gcc
+              os: ubuntu-latest
             cpp_version: 17
             cmake_args:
               description: "Dynamic"
               args: "-DBUILD_SHARED_LIBS=on"
+            # MSVC has limited support for sanitizers, so we use include here for MSVC as an exception.
+            # This should be simplified after beman-project/exemplar#76 is merged.
+          - platform:
+              description: "windows MSVC"
+              cpp: cl
+              c: cl
+              os: windows-latest
+            cpp_version: 17
+            cmake_args:
+              description: "Default"
+              args: ""
+          - platform:
+              description: "windows MSVC"
+              cpp: cl
+              c: cl
+              os: windows-latest
+            cpp_version: 17
+            cmake_args:
+              description: "ASan"
+              # Debug infomation needed to avoid cl: C5072
+              # https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-c5072?view=msvc-170
+              args: "-DCMAKE_CXX_FLAGS='/fsanitize=address /Zi'"
+
 
-    name: "Unit: ${{ matrix.compiler.c }} ${{ matrix.cpp_version }} ${{ matrix.cmake_args.description }}"
-    runs-on: ${{ matrix.platform }}
+    name: "Unit: ${{ matrix.platform.description }} ${{ matrix.cpp_version }} ${{ matrix.cmake_args.description }}"
+    runs-on: ${{ matrix.platform.os }}
     steps:
       - uses: actions/checkout@v4
       - name: Install Ninja
         uses: lukka/get-cmake@latest
         with:
           cmakeVersion: "~3.25.0"
           ninjaVersion: "^1.11.1"
+      - name: Setup MSVC
+        if: startsWith(matrix.platform.os, 'windows')
+        uses: TheMrMilchmann/setup-msvc-dev@v3
+        with:
+          arch: x64
       - name: Print installed softwares
         run: |
-          clang++ --version
-          g++ --version
+          echo "Compiler:"
+          ${{ matrix.platform.cpp }} --version
+          ${{ matrix.platform.c }} --version
```

Comment:
Note that `--version` isn't a `cl.exe` flag. Maybe something like this? See [this reference page](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#operators).

```suggestion
          ${{ matrix.platform.cpp }} ${{ matrix.platform.cpp == 'cl' && '' || '--version' }}
          ${{ matrix.platform.c }}  ${{ matrix.platform.cpp == 'cl' && '' || '--version' }}
```

Reply:
I don't think this is working.

Also `--version` seem to work? (I know this shouldn't but it's working).
![image](https://github.com/user-attachments/assets/d98f9f2c-56cf-4691-b1b9-4e7484eb42a8)

Reply:
done.